src/linker/utilities/spark_utils.py

```diff
@@ -151,8 +159,17 @@ def find_spark_master_url(logfile: Path) -> str:
     spark_master_url = ""
     while spark_master_url == "":
         sleep(5)
-        with open(logfile, "r") as f:
-            for line in f:
-                if "Starting Spark master at" in line:
-                    spark_master_url = line.split(" ")[-1:]
+        try:
+            with open(logfile, "r") as f:
+                for line in f:
+                    if "Starting Spark master at" in line:
+                        spark_master_url = line.split(" ")[-1:]
+        except FileNotFoundError:
+            logger.warning(f"Logfile {logfile} not found. Waiting 5 seconds and retrying...")
+            sleep(5)
+            with open(logfile, "r") as f:
+                for line in f:
+                    if "Starting Spark master at" in line:
+                        spark_master_url = line.split(" ")[-1:]
```

Comment:
```suggestion
            continue
```

Reply:
Or did you want it to only retry once? Seems dangerous.

Reply:
So frankly I'm surprised I needed to sleep at all. Most of the time the file is there and there are no issues. So whatever race condition exists seems to be short and I figured a single 5-sec sleep was enough. But this is certainly cleaner. I think I'll add an upper limit though just in case something is truly preventing the file from being written